app/models/authn_oidc.rb

```diff
@@ -0,0 +1,108 @@
+class AuthnOidc
+
+  def authenticators(role, account:, service_id: nil)
+    repo = AuthenticatorRepo.new(role, account, service_id)
+    variables = repo.fetch_authenticators
+    return list_authenticators(variables) unless service_id
+
+    find_service(variables, service_id)
+  end
+
+  def list_authenticators(variables)
+    args_list = []
+    variables.each do |variable|
+      args = {}
+      args[:service_id] = variable.owner_id.split('/')[-1].underscore.to_sym
+      args[variable.resource_id.split('/')[-1].underscore.to_sym] =
+        variable.secret.value
+      args_list.push(args)
+    end
+    args_list.group_by{|authn| authn[:service_id]}.map do |_, authn|
+      Authenticator.new(**authn.reduce({}, :merge)).return_service
+    end
+  end
+
+  private
+
+  def find_service(variables, service_id)
+    return {} unless variables.count >=1
+
+    args_list = {}.tap do |args|
+      args[:service_id] = service_id
+      variables.each do |variable|
+        args[variable.resource_id.split('/')[-1].underscore.to_sym] =
+          variable.secret.value
+      end
+    end
+    Authenticator.new(**args_list).return_uri
+  end
+
+  class Authenticator
```

Comment:
Please try to avoid this "class in class" construction.  If classes are related, place them into a `module`.  This allows us to keep them in separate files, which forces discipline around single responsibility.